File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        char largestNum = 0;
        string goodStr = "";
        for(int i = 0; i < num.length() - 2; i++)
        {
            if(num[i] == num[i + 1] && num[i + 1] == num[i + 2] && num[i] == num[i + 2])
            {
                if(goodStr.length() == 0 || num[i] - '0' > largestNum)
                {
                    goodStr = num.substr(i, 3);
                    largestNum = num[i] - '0';
                }
            }
        }
        return goodStr;
    }
};

File: 2346-largest-3-same-digit-number-in-string/2346-largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        largestNum = 0
        goodIdx = -1
        for i in range(len(num) - 2):
            if (
                num[i] == num[i + 1]
                and num[i + 1] == num[i + 2]
                and num[i] == num[i + 2]
            ):
                if ord(num[i]) - ord("0") > largestNum or goodIdx == -1:
                    goodIdx = i
                    largestNum = ord(num[i]) - ord("0")
        return num[goodIdx : goodIdx + 3]
